comms/discovery/db/queries/get_chat_blocked_users.go

```diff
@@ -35,3 +36,46 @@ func CountChatBlocks(q db.Queryable, ctx context.Context, arg CountChatBlocksPar
 	err := q.GetContext(ctx, &count, countChatBlocks, arg.User1, arg.User2)
 	return count, err
 }
+
+const bulkGetChatBlocksOrReceivedBlocks = `
+select blocker_user_id, blockee_user_id from chat_blocked_users where (blocker_user_id = :SenderUserID and blockee_user_id in (:ReceiverUserIDs)) or (blockee_user_id = :SenderUserID and blocker_user_id in (:ReceiverUserIDs))
+`
+
+type BulkGetChatBlocksOrReceivedBlocksParams struct {
+	SenderUserID    int32   `json:"sender_user_id"`
+	ReceiverUserIDs []int32 `json:"receiver_user_ids"`
```

Comment:
Also just thought of this - we could potentially have a way for users to get _all_ users that block them, and the front end could call `getBlockers` and `getBlockees` _once_ rather than for each batch of users in the search users modal, and we'd only have to get chat permission stuff on demand? (def not blocking this PR for that, just musing - let's get this in!)

Reply:
yeah i like it; would be fewer queries to the db if we don't mind moving the validation logic to the frontend for search. we should rename the validate-can-chat endpoint to be permissions specific in that case for clarity also

Reply:
I suspect down the road this will become more necessary to understand _why_ a user isn't allowed to chat another user. Having permissions and blocks separate could help with that

Reply:
we will already have queried `/chats/blocked-users` at the beginning of the user's session as well